File: apps/bot/commands/music/join.ts
import { joinVoiceChannel } from "@discordjs/voice";
import { DiscordCommand } from "../../types/command";

export const command: DiscordCommand = {
    name: "join",
    category: "music",
    description: "Joins the voice channel you are in.",
    run: async (client, message, args) => {
        const channel = message.member?.voice.channel;
        if (!channel) return message.reply("You must be in a voice channel to use this command.");

        if (client.audio.connection) return message.reply("I am already in a voice channel.");

        client.audio.channel = channel;
        client.audio.connection = joinVoiceChannel({
            channelId: channel.id,
            guildId: channel.guild.id,
            adapterCreator: channel.guild.voiceAdapterCreator
        });

        message.reply(`Joined ${channel.name}.`);
    }
}

File: bot/commands/join.go
package commands

import (
	"github.com/bwmarrin/discordgo"
	"github.com/lofustudio/VEGA/bot/voice"
	"github.com/rs/zerolog/log"
	"go.etcd.io/bbolt"
)

type JoinCommand struct{}

func (p JoinCommand) Name() string {
	return "join"
}

func (p JoinCommand) Description() string {
	return "Join the voice channel you are in!"
}

func (p JoinCommand) Run(s *discordgo.Session, m *discordgo.MessageCreate, _ *bbolt.DB) {
	authorVCID, err := voice.DetectVoiceChannel(s, m)
	if err != nil {
		log.Error().Err(err).Msg("Could not find the author's vc")
		_, err := s.ChannelMessageSendReply(m.ChannelID, "Couldn't find the voice channel!", m.Reference())
		if err != nil {
			log.Error().Err(err).Msg("Failed to send reply")
			return
		}
		return
	}

	_, err = voice.JoinVoice(s, m.GuildID, authorVCID)
	if err != nil {
		log.Error().Err(err).Msg("Could not join the author's vc")
		_, err := s.ChannelMessageSendReply(m.ChannelID, "Couldn't join the voice channel!", m.Reference())
		if err != nil {
			log.Error().Err(err).Msg("Failed to send reply")
			return
		}
		return
	}
}

var _ Command = (*JoinCommand)(nil)
